barretenberg/cpp/src/barretenberg/common/compiler_hints.hpp

```diff
@@ -23,4 +23,6 @@
 #else
 #define BB_LIKELY(x) x
 #define BB_UNLIKELY(x) x
-#endif
\ No newline at end of file
+#endif
+
+#define BB_UNUSED [[maybe_unused]]
```

Comment:
I think this is a good idea but its worth adding a comment for when people say what the hell is this

Reply:
I'll smuggle this next PR